update.go

```diff
@@ -0,0 +1,41 @@
+package vaultrole
+
+import (
+	"github.com/giantswarm/microerror"
+)
+
+func (r *VaultRole) Update(config UpdateConfig) error {
+	// Check if the requested role exists.
+	{
+		c := ExistsConfig{
+			ID:            config.ID,
+			Organizations: config.Organizations,
+		}
+		exists, err := r.Exists(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if !exists {
+			return microerror.Maskf(notFoundError, "cannot update Vault role '%s'", config.ID)
+		}
+	}
+
+	// Update the requested role if it exists.
+	{
+		c := writeConfig{
+			AllowBareDomains: config.AllowBareDomains,
+			AllowSubdomains:  config.AllowSubdomains,
+			AltNames:         config.AltNames,
+			ID:               config.ID,
+			Organizations:    config.Organizations,
+			TTL:              config.TTL,
+		}
+
+		err := r.write(c)
```

Comment:
Same as in `VaultRole.Create`, we use the common method now.

Reply:
I don't like that. Why not `EnsureCreated` like in CRDClient?

Reply:
I want to have this for the operator resource. When we want to create something we create something. In case something exists already we get an error and can be more sure what we are doing. Same principle for updating but the other way around with the existence check.